services/frontend-service/src/ui/components/LockDisplay/LockDisplay.test.tsx

```diff
@@ -117,3 +122,76 @@ describe('Test Auxiliary Functions for Lock Display', () => {
         });
     });
 });
+
+describe('Test delete lock button', () => {
+    interface dataT {
+        name: string;
+        actions: BatchAction[];
+        expectedNumOfActions: number;
+        date: Date;
+    }
+    const lock = {
+        date: new Date('2/1/22'),
+        environment: 'test-env',
+        // application?: string,
+        message: 'string',
+        lockId: 'test-lock-id',
+        authorName: 'test-auth',
+        authorEmail: 'test-mail',
+    };
+    const data: dataT[] = [
+        {
+            name: 'Test environment delete button',
+            date: new Date('2/1/22'),
+            actions: [
+                {
+                    action: {
+                        $case: 'createEnvironmentLock',
+                        createEnvironmentLock: {
+                            environment: 'nmww',
+                            lockId: 'test-lock-id',
+                            message: 'test-message',
+                        },
+                    },
+                },
+                {
+                    action: {
+                        $case: 'createEnvironmentApplicationLock',
+                        createEnvironmentApplicationLock: {
+                            environment: 'nmww',
+                            application: 'test-application',
+                            lockId: 'test-lock-id',
+                            message: 'test-message',
+                        },
+                    },
+                },
+            ],
+            expectedNumOfActions: 1,
+        },
+    ];
+
+    describe.each(data)('', (testcase) => {
+        it('Cart initially empty', () => {
+            render(<LockDisplay lock={lock} />);
+            expect(document.getElementsByClassName('mdc-drawer-sidebar-list').length).toBe(0);
+        });
+        it(testcase.name, () => {
+            const api = useApi;
+            // given
+            updateActions(testcase.actions);
+            api.batchService().ProcessBatch(testcase.actions);
```

Comment:
please don't use the api here. you can use the overview store and fill it with locks.
But that's not even required. the LockDisplay does not read anything directly from the store.